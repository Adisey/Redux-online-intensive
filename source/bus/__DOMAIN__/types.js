export const type = {
    TYPE:        'TYPE',
};
