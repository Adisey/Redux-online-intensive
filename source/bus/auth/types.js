export const type = {
    // Sync
    AUTHENTICATE:       'AUTHENTICATE',
    // Async
    SIGNUP_ASYNC:       'SIGNUP_ASYNC',
    LOGIN_ASYNC:        'LOGIN_ASYNC',
    AUTHENTICATE_ASYNC: 'AUTHENTICATE_ASYNC',
};
