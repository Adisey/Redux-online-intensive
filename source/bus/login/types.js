export const type = {
    // Sync
    AUTHENTICATEL: 'AUTHENTICATEL',
    // Async
    LOGIN_ASYNC: '',
};