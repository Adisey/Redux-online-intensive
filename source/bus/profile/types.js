export const type = {
    FILL_PROFILE: 'FILL_PROFILE',
    CLEAR_PROFILE: 'CLEAR_PROFILE',
};
