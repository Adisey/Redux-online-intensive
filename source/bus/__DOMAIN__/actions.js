import { type } from "./types";

export const postsActions ={
    action: () => {
        return {
            type:    type.TYPE,
        };
    },
};
