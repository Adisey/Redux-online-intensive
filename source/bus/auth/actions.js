import { type } from "./types";

export const authAction ={
    //Sync
    authenticate: () => {
        return {
            type: type.AUTHENTICATE,
        };
    },
    // Async
    signupAsync: (userData) => {
        return {
            type:    type.SIGNUP_ASYNC,
            payload: userData,
        };
    },
    loginAsync: (userData) => {
        console.log(`userData ->`, userData);
        return {
            type:    type.LOGIN_ASYNC,
            payload: userData,
        };
    },
};
