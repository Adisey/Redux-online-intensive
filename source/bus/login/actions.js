import { type } from "./types";

export const loginActions ={
    //Sync
    authenticatel: () => {
        return {
            type: type.AUTHENTICATEL,
        };
    },
    // Async
    loginAsync: (userData) => {
        console.log(`userrData ->`, userData);
        return {
            type:    type.LOGIN_ASYNC,
            payload: userData,
        };
    },
};

