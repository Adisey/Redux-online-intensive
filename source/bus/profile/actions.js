import { type } from "./types";

export const profileActions ={
    // Sync
    fillProfile: (profile) => {
        return {
            type:    type.FILL_PROFILE,
            payload: profile,
        };
    },
    clearProfile: () => {
        return {
            type: type.CLEAR_PROFILE,
        };
    },
    // Async
    updateNameAsync: (newName) => {
        return {
            type:    type.UPDATE_NAME_ASYNC,
            payload: newName,
        };
    },
};
