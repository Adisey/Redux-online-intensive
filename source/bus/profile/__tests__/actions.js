/**
 * Created by PhpStorm
 * Project p501-redux-online-intensive
 * User: Adisey
 * Date: 28.08.2018
 * Time: 17:33
 */
// Actions
import { profileActions } from '../actions';
// Types
import { type } from '../types';

describe('Экшены Profile:', () => {
    test('fillProfile Snapshot', () => {
        expect(profileActions.fillProfile(__.userProfile))
            .toMatchSnapshot();
    });

    test('fillProfile', () => {
        expect(profileActions.fillProfile(__.userProfile))
            .toEqual({
                type:    type.FILL_PROFILE,
                payload: __.userProfile,
            });
    });

    test('updateAvatar Snapshot 1 ', () => {
        expect(profileActions.updateAvatar(__.newAvatarUrl))
            .toMatchSnapshot();
    });

    test('updateAvatar Snapshot 2 ', () => {
        expect(profileActions.updateAvatar(__.url))
            .toMatchSnapshot();
    });

    test('updateAvatar', () => {
        expect(profileActions.updateAvatar(__.newAvatarUrl))
            .toEqual({
                type:    type.UPDATE_AVATAR,
                payload: __.newAvatarUrl,
            });
    });

    test('clearProfile Snapshot', () => {
        expect(profileActions.clearProfile())
            .toMatchSnapshot();
    });

    test('clearProfile', () => {
        expect(profileActions.clearProfile())
            .toEqual({
                type: type.CLEAR_PROFILE,
            });
    });

    test('updateNameAsync', () => {
        expect(profileActions.updateNameAsync(__.userProfile.firstName))
            .toEqual({
                type:    type.UPDATE_NAME_ASYNC,
                payload: __.userProfile.firstName,
            });
    });

    test('updateNameAsync Snapshot', () => {
        expect(profileActions.updateNameAsync(__.newName))
            .toMatchSnapshot();
    });

    test('updateAvatarAsync', () => {
        expect(profileActions.updateAvatarAsync(__.newAvatarUrl))
            .toEqual({
                type:    type.UPDATE_AVATAR_ASYNC,
                payload: __.newAvatarUrl,
            });
    });

    test('updateAvatarAsync Snapshot', () => {
        expect(profileActions.updateAvatarAsync(__.newAvatar))
            .toMatchSnapshot();
    });

    test('updatePasswordAsync', () => {
        expect(profileActions.updatePasswordAsync(__.userProfile.firstName))
            .toEqual({
                type:    type.UPDATE_PASSWORD_ASYNC,
                payload: __.userProfile.firstName,
            });
    });

    test('updatePasswordAsync Snapshot', () => {
        expect(profileActions.updatePasswordAsync(__.userProfile.firstName))
            .toMatchSnapshot();
    });

});
