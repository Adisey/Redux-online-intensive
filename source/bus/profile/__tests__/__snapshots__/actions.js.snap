// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Экшены Profile: clearProfile Snapshot 1`] = `
Object {
  "type": "CLEAR_PROFILE",
}
`;

exports[`Экшены Profile: fillProfile Snapshot 1`] = `
Object {
  "payload": Object {
    "avatar": "TEST_AVATAR",
    "firstName": "Walter",
    "id": "TEST_ID",
    "lastName": "White",
    "token": "TEST_TOKEN",
  },
  "type": "FILL_PROFILE",
}
`;

exports[`Экшены Profile: updateAvatar Snapshot 1  1`] = `
Object {
  "payload": "NEW_AVATAR_URL",
  "type": "UPDATE_AVATAR",
}
`;

exports[`Экшены Profile: updateAvatar Snapshot 2  1`] = `
Object {
  "payload": "https://www.url.com",
  "type": "UPDATE_AVATAR",
}
`;

exports[`Экшены Profile: updateAvatarAsync Snapshot 1`] = `
Object {
  "payload": Array [
    "avatar",
  ],
  "type": "UPDATE_AVATAR_ASYNC",
}
`;

exports[`Экшены Profile: updateNameAsync Snapshot 1`] = `
Object {
  "payload": Object {
    "firstName": "Walter",
    "lastName": "White",
  },
  "type": "UPDATE_NAME_ASYNC",
}
`;

exports[`Экшены Profile: updatePasswordAsync Snapshot 1`] = `
Object {
  "payload": "Walter",
  "type": "UPDATE_PASSWORD_ASYNC",
}
`;
